api/envoy/config/core/v3/base.proto

```diff
@@ -329,7 +329,24 @@ message HeaderValue {
   ];
 }
 
-// Header name/value pair plus option to control append behavior.
+// Header name/value pair with header value as bytes.
+message HeaderValueBytes {
+  option (udpa.annotations.versioning).previous_message_type = "envoy.api.v2.core.HeaderValue";
```

Comment:
This option is redundant? As you are adding a new message here

Reply:
We are using oneof approach in the HeaderValue message. That approach make it possible to make this PR a non-breaking change.  With that, the new sets of messages are removed.